 <!-- async / await -->
<script>
    var checknum = (num)=>{
        return new Promise((resolve,reject)=>{
            (num%2==0) ? setTimeout(()=>{resolve("Even Number")},2000) : reject("Odd Number");
        });
    }
    console.log("before async");
    async function result(){
        console.log("before try");
        try{
            console.log("before execution");
                var res = await checknum(100);
                console.log("--Message : "+res);
            console.log("after execution");
        }catch(err){
            document.write("--Error : "+err);
        }    
        console.log("after catch");
    }
    console.log("after async");
    result();
    console.log("after result");


    // checknum(100).then((msg)=>{
    //     document.write("Message : "+msg);
    // }).catch((msg)=>{
    //     document.write("Message : "+msg);
    // });
    
</script>